Creation of Edge Resource Mananger.

petedyer11 opened the pull request:
The Edge Resource manager adds LwM2M resources to edge-core using the gateway resource manager REST API.

https://developer.izumanetworks.com/docs/device-management-edge/2.6/managing/grm-json-rpc.html 

The resources to be added are specified in a config file.

Commit:
Creation of Edge Resource Mananger.

The Edge Resource manager adds LwM2M resources to edge-core using the
gateway resource manager REST API.
https://developer.izumanetworks.com/docs/device-management-edge/2.6/managing/grm-json-rpc.html
The resources to be added are specified in a config file.

JanneKiiskila commented on resourcemanager/resource_manager.go:
```diff
@@ -0,0 +1,287 @@
+    }
+    var err error
+    gcdConfig = config
+    log.Infof("ResourceManager: connecting to edge-core\n")
```
Do you really need to add the  `\n`'s there? Could the linechanges be behind some API instead?

JanneKiiskila commented on README.md:
```diff
@@ -1 +1,40 @@
\ No newline at end of file
+# Edge Resource Manager
+
+The Edge Resource Manager adds LwM2M resources to an Edge device at runtime
```
```suggestion
The Edge Resource Manager adds LwM2M resources to an Edge device at runtime.
```

JanneKiiskila commented on README.md:
```diff
@@ -1 +1,40 @@
+
+## Configuration
+
+The edge-resource manager takes a configuration file as it's input that describes the resources that are to be added to edge-core.
```
```suggestion
The edge-resource manager takes a configuration file as its input that describes the resources that are to be added to edge-core.
```

JanneKiiskila commented on README.md:
```diff
@@ -1 +1,40 @@
+
+The edge-resource-manager should be started as a systemd service on the gateway.
+
+A typical service file is
```
```suggestion
A typical service file is shown below:
```

JanneKiiskila reviewed (changes requested):
Small changes. Have not had a chance to review the go-code in full yet.

JanneKiiskila commented on .github/workflows/static-checks.yml:
```diff
@@ -0,0 +1,73 @@
+        run: |
+            # Remove the vendor folder - we don't need to check that.
+            rm -rf vendor
+            curl -L -o ./install-misspell.sh https://git.io/misspell
```
There's a different repo we should pull this from. A more maintained version is in:  https://github.com/golangci/misspell

Commit:
Update README.md

Co-authored-by: Janne Kiiskilä <janne.kiiskila@izumanetworks.com>

Commit:
Update README.md

Co-authored-by: Janne Kiiskilä <janne.kiiskila@izumanetworks.com>

Commit:
Update README.md

Co-authored-by: Janne Kiiskilä <janne.kiiskila@izumanetworks.com>

Commit:
Address PR comments.

Updated version of misspell.

Commit:
Address PR comments.

Updated version of misspell.

Commit:
Address PR comments.

Simlified action - no need to include shellcheck when there are no
shell files.

Commit:
Address PR comments.

Removed unneccesary '\n' from log messages.
